app/storage/src/main/java/com/fsck/k9/storage/migrations/MigrationTo84.kt

```diff
@@ -0,0 +1,54 @@
+package com.fsck.k9.storage.migrations
+
+import android.content.ContentValues
+import android.database.sqlite.SQLiteDatabase
+import com.fsck.k9.helper.map
+import com.fsck.k9.mailstore.MigrationsHelper
+
+/**
+ * Write the address fields to use ASCII 1 instead of ASCII 0 as separator.
+ * Separator was previously ASCII 0 but this caused problems with LIKE and searching.
+ */
+internal class MigrationTo84(private val db: SQLiteDatabase, private val migrationsHelper: MigrationsHelper) {
+
+    fun rewriteAddresses() {
+        val addressSets = db.rawQuery(
+            "SELECT id, to_list, cc_list, bcc_list, reply_to_list, sender_list FROM messages",
```

Comment:
We want to reduce the number of rows to update as much as possible. If you add a WHERE-clause of `WHERE empty = 0 AND deleted = 0` we skip rows we know will contain NULL in all of the address columns.